### ISSUE TITLE
Pilot fails with `{"code"=>"Enter a demo account user name, or deselect` since update from 2.72.0 > 2.78.0

### ISSUE THREAD

**Issue by Blackjacx** [fastlane/fastlane#11665]
### Issue Description

Since I updated to fastlane 2.78  I get the error above and my CI build fails. But the app is uploaded. The error is triggered during distributing the build to testers.

##### Complete output when running fastlane, including the stack trace and command used
```
[07:48:55]: Distributing new build to testers: 0.20.0 - 3600
[07:49:10]: Sending crash report...
[07:49:10]: The stack trace is sanitized so no personal information is sent.
[07:49:10]: To see what we are sending, look here: /Users/vagrant/.fastlane/latest_crash.json
[07:49:10]: Learn more at https://docs.fastlane.tools/actions/opt_out_crash_reporting/
[07:49:10]: You can disable crash reporting by adding `opt_out_crash_reporting` at the top of your Fastfile
[07:49:10]: -------------------
[07:49:10]: --- Step: slack ---
[07:49:10]: -------------------
[07:49:10]: Successfully sent Slack notification
[07:49:10]: {"code"=>"Enter a demo account user name, or deselect \"A demo account is required to use all the features of the app.\"", "description"=>"Failed to submit build for review."}
+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | Verifying fastlane version | 0           |
| 2    | default_platform           | 0           |
| 3    | cocoapods                  | 70          |
| 4    | get_version_number         | 6           |
| 5    | Switch to ios test lane    | 0           |
| 6    | scan                       | 511         |
| 7    | danger                     | 1           |
| 8    | increment_version_number   | 1           |
| 9    | increment_build_number     | 2           |
| 10   | read_changelog             | 0           |
| 11   | gym                        | 844         |
| 12   | set_github_release         | 0           |
| 13   | crashlytics                | 54          |
| 💥   | pilot                      | 1748        |
| 15   | slack                      | 0           |
+------+----------------------------+-------------+
[07:49:10]: fastlane finished with errors
Looking for related GitHub issues on fastlane/fastlane...
Found no similar issues. To create a new issue, please visit:
https://github.com/fastlane/fastlane/issues/new
Run `fastlane env` to append the fastlane environment to your issue
bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
Spaceship::UnexpectedResponse: [!] {"code"=>"Enter a demo account user name, or deselect \"A demo account is required to use all the features of the app.\"", "description"=>"Failed to submit build for review."}
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/spaceship/lib/spaceship/test_flight/client.rb:290:in `handle_response'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/spaceship/lib/spaceship/test_flight/client.rb:72:in `post_for_testflight_review'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/spaceship/lib/spaceship/test_flight/build.rb:204:in `submit_for_testflight_review!'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/pilot/lib/pilot/build_manager.rb:149:in `distribute_build'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/pilot/lib/pilot/build_manager.rb:88:in `distribute'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/pilot/lib/pilot/build_manager.rb:52:in `upload'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/runner.rb:231:in `block in execute_action'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
  Fastfile:180:in `block (2 levels) in parsing_binding'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
  /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
  /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
  /usr/local/lib/ruby/gems/2.4.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/commands_generator.rb:337:in `run'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/fastlane/lib/fastlane/cli_tools_distributor.rb:98:in `take_off'
  /usr/local/lib/ruby/gems/2.4.0/gems/fastlane-2.78.0/bin/fastlane:20:in `<top (required)>'
  /usr/local/bin/fastlane:23:in `load'
  /usr/local/bin/fastlane:23:in `<top (required)>'
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.2                                     |
| Ruby                        | 2.4.2                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.15.1                          |
| Installation Source         | ~/.gem/bin/fastlane                         |
| Host                        | Mac OS X 10.13.2 (17C205)                   |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.4.2_1/lib          |
| OpenSSL Version             | OpenSSL 1.0.2l  25 May 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Fastlane Documentation:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
#
# Fastlane actions: 
# https://docs.fastlane.tools/actions/
#
# Bitrise with fastlane example:
# https://github.com/bitrise-samples/fastlane/blob/master/BitriseFastlaneSample/fastlane/Fastfile
#
# AUTHENTICATION
# https://docs.fastlane.tools/best-practices/continuous-integration/
# https://github.com/fastlane/fastlane/tree/master/spaceship
#
# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`
#
#
# This file was created by the help of https://gist.github.com/ulhas/e8e529d08849b8cda947
#
#
# NOTE: To use faslane first do:
#           gem install bundler && bundle install
#
#       and then run:
#           bundle exec fastlane release product_name:Gasoline version:"1.0.0" github_account:"blackjacx"
#           bundle exec fastlane test product_name:Gasoline
#
# Configure the following environment variables for your project 
# on your CI and in fastlane/.env:
#
# ENV["GITHUB_TOKEN"] = "..."
# ENV["SLACK_URL"] = "..."
# ENV["BITRISE_DEPLOY_DIR"] = "..."
# ENV['CRASHLYTICS_API_TOKEN'] = "..."
# ENV['CRASHLYTICS_BUILD_SECRET'] = "..."
# ENV['FASTLANE_USER'] = "..."
# ENV['FASTLANE_PASSWORD'] = "..."
# ENV['FASTLANE_DONT_STORE_PASSWORD'] = "..."



# TODO: deploy to TestFlight, Crashlytics, Hockey, ... & move away from Bitrise



# This is the minimum version number required. 
# Update this, if you use features of a newer version.
fastlane_version "2.55.0"

default_platform :ios


platform :ios do

  before_all do
    cocoapods(use_bundle_exec: true, repo_update: true)
  end

  after_all do
    # This is called, only if the executed lane was successful
    slack(success: true, message: "*Successfully Built App 🎉*")
  end

  error do |lane, exception|
    # This block is called, only if the executed lane failed
    slack(success: false, message: "*#{lane} failed with #{exception.message}*")
  end

  ##############################################################################
  # Playground Lane
  ##############################################################################

  lane :playground do |options|
    stamp_changelog(
      section_identifier: version, # Stamp Unreleased section with newly released build number
      placeholder_line: "* Your contribution here."
    )
  end

  ##############################################################################
  # Preparing Release
  ##############################################################################

  lane :prepare_release do |options|

    raise "No product_name provided!".red unless options[:product_name]
    raise "No version provided!".red unless options[:version]

    product_name = options[:product_name]
    version = options[:version]

    ensure_git_status_clean
    ensure_git_branch(branch: "develop")

    increment_version_number(
      version_number: version
    )

    stamp_changelog(
      section_identifier: version, # Stamp Unreleased section with newly released build number
      placeholder_line: "* Your contribution here."
    )
  end

  ##############################################################################
  # Release Lane
  ##############################################################################

  desc "Releases a new product version"
  lane :release do |options|

    possible_tester_groups = ["appstore", "beta", "nightly"]
    tester_groups = "#{options[:tester_groups]}"

    raise "No product_name provided!".red unless options[:product_name]
    raise "No build provided!".red unless options[:build]
    raise "No github_account provided!".red unless options[:github_account]
    raise "No tester groups provided! Provide at least one of the following as ruby array: #{possible_tester_groups}".red if options[:tester_groups].empty?
    
    product_name = options[:product_name]
    build = options[:build]
    github_account = options[:github_account]
    project = "#{product_name}.xcodeproj"
    workspace = "#{product_name}.xcworkspace"
    version = get_version_number(xcodeproj: "#{project}", target: "#{product_name}")
    deploy_dir = ENV["BITRISE_DEPLOY_DIR"]
    ipa_name = "#{product_name}.ipa"
    ipa_path = "#{deploy_dir}/#{product_name}.ipa"
    dsym_path = "#{deploy_dir}/#{product_name}.app.dSYM.zip"
    run_danger = options[:run_danger]
    tester_whatsnew = File.read("./metadata/whatsnew.txt")
    export_options_plist = "./fastlane/ExportOptions.plist"

    test(product_name: product_name, run_danger: run_danger)

    # Increments CFBundleVersion by one and sets CFBundleShortVersionString
    increment_version_number(version_number: version)
    increment_build_number(build_number: build)

    # read changelog
    changelog = read_changelog(
      section_identifier: "[#{version}]",
      excluded_markdown_elements: []
    )

    changelog_no_md = changelog
        .gsub(/\* \[(.+): /, '* ')
        .gsub(/ - (.+)\./, '')
        .gsub(/###/, '')
        .strip

    gym(
      workspace: workspace, 
      scheme: product_name,
      buildlog_path: deploy_dir,
      output_directory: deploy_dir,
      output_name: ipa_name,
      export_options: export_options_plist,
      include_bitcode: true,
      include_symbols: true,
      clean: true
    )

    set_github_release(
      repository_name: "#{github_account}/#{product_name}",
      api_token: ENV["GITHUB_TOKEN"],
      name: version,
      tag_name: version,
      description: (changelog || "No changelog provided"),
      commitish: "master"
    )
    
    crashlytics(
      crashlytics_path: "./Pods/Crashlytics/iOS/Crashlytics.framework",
      groups: tester_groups,
      notes: changelog_no_md,
      notifications: false
    )

    pilot(
      changelog: changelog_no_md,
      ipa: "#{ipa_path}",
      groups: tester_groups,
      distribute_external: true
    )

    sh("ls -la #{deploy_dir}")
    puts tester_groups
  end

  ##############################################################################
  # Test Lane
  ##############################################################################

  desc "Runs tests optionally with danger"
  lane :test do |options|

    raise "No product_name provided!".red unless options[:product_name]

    product_name = options[:product_name]
    run_danger = options[:run_danger]
    
    # Easily run tests of your iOS app using scan
    scan(
      workspace: "#{product_name}.xcworkspace",
      scheme: product_name,
      clean: true,
      devices: ["iPhone 8"],
      skip_build: false,
      output_types: "html",
      skip_slack: true,
      output_directory: ENV["BITRISE_DEPLOY_DIR"]
    )

    # Runs danger for the project
    if run_danger
      danger(
        use_bundle_exec: true,
        github_api_token: ENV["GITHUB_TOKEN"],
        verbose: true
      )
    end
  end

  ##############################################################################
  # Manually Refreshing Crashlytics dSYM's
  ##############################################################################

  lane :refresh_dsyms do
    # This needs to be done after uploading a new app to ITC since Apple 
    # recompiles the apps from bitcode hich results in new dSYM's.
    # NOTE: upload_symbols_to_crashlytics relies in the installed Fabric.app
    # https://github.com/fastlane/fastlane/issues/10255
    # https://krausefx.com/blog/download-dsym-symbolication-files-from-itunes-connect-for-bitcode-ios-apps
    # https://docs.fabric.io/apple/crashlytics/missing-dsyms.html#upload-symbols-script
    user = ENV["FASTLANE_USER"]
    app_id = CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier)
    
    download_dsyms(username: user, app_identifier: app_id)
    upload_symbols_to_crashlytics
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "com.ioki.passenger" # The bundle identifier of your app

team_id "P4LECE5B35" # Developer Portal Team ID
itc_team_id "118614410" # ID identifying the team Deutsche Bahn AG

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.78.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                    | Version | Update-Status |
| ------------------------- | ------- | ------------- |
| fastlane-plugin-changelog | 0.9.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| bundler                   | 1.16.0       |
| io-console                | 0.4.6        |
| rake                      | 12.3.0       |
| CFPropertyList            | 2.3.6        |
| concurrent-ruby           | 1.0.5        |
| i18n                      | 0.9.3        |
| minitest                  | 5.11.1       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.4        |
| activesupport             | 4.2.10       |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| cork                      | 0.3.0        |
| nap                       | 1.1.0        |
| open4                     | 1.3.4        |
| claide-plugins            | 0.9.2        |
| fuzzy_match               | 2.0.4        |
| cocoapods-core            | 1.4.0        |
| cocoapods-deintegrate     | 1.0.2        |
| cocoapods-downloader      | 1.1.3        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.11.0       |
| cocoapods-trunk           | 1.3.0        |
| cocoapods-try             | 1.1.0        |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.0.3        |
| molinillo                 | 0.6.4        |
| ruby-macho                | 1.1.0        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.4        |
| cocoapods                 | 1.4.0        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| multipart-post            | 2.0.0        |
| faraday                   | 0.14.0       |
| faraday-http-cache        | 1.3.1        |
| git                       | 1.3.0        |
| kramdown                  | 1.16.2       |
| no_proxy_fix              | 0.1.2        |
| sawyer                    | 0.8.1        |
| octokit                   | 4.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| danger                    | 5.5.6        |
| danger-plugin-api         | 1.0.0        |
| danger-changelog          | 0.2.1        |
| danger-commit_lint        | 0.0.6        |
| thor                      | 0.20.0       |
| danger-swiftlint          | 0.12.1       |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.1        |
| excon                     | 0.60.0       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| fastimage                 | 2.1.1        |
| jwt                       | 2.1.0        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.13.1       |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| google-api-client         | 0.13.6       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.4.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 1.8.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| xcpretty-travis-formatter | 1.0.0        |
| fastlane-plugin-changelog | 0.9.0        |
</details>

*generated on:* **2018-01-23**
</details>

**Comment by alakoring** [comment]
Same here. It is apparently unable to uncheck the demo account (that I had been doing manually).

**Comment by alakoring** [comment]
Expected behavior is that if I don’t specify a demo user, that TestFlight is told there is none.

**Comment by Blackjacx** [comment]
The question is since there is an option under the TestFlight tab to specify/deactivate a demo user, why do I need to do this per build version again? OK, I have sth in mind which is you built in login in a later version, which means you wouldn't need a demo user from version 0 but let's say from 1.5.0. But then I don't understand why ITC does not respect the global setting from TestFlight and just does not show this dialog!?

**Comment by joshdholtz** [comment]
Should be fixed with now with #11760 😊 Happy fastlaning 🚀
